### PR TITLE
initial effort to remove numpy dependent in compiling time.

### PR DESCRIPTION
**Pull request by hainm**

### CODE REVIEW

File: package/MDAnalysis/lib/distances_parallel.pyx [hunk 1]
```diff
@@ -78,26 +78,21 @@ Load the module with ::
    .. versionadded:: 0.7.7
```
Comment (richardjgowers) [inline package/MDAnalysis/lib/distances_parallel.pyx]: We're removing this module in #530

File: package/MDAnalysis/lib/src/pyqcprot/pyqcprot.pyx [hunk 2]
```diff
@@ -136,11 +134,11 @@ cdef extern from "math.h":
 
 @cython.boundscheck(False)
 @cython.wraparound(False)
-def InnerProduct(np.ndarray[np.float64_t,ndim=1] A,
```
Comment (richardjgowers) [inline package/MDAnalysis/lib/src/pyqcprot/pyqcprot.pyx]: This syntax is much nicer imo

File: package/MDAnalysis/lib/distances_parallel.pyx (hunk 1 above)
Comment (hainm) [inline package/MDAnalysis/lib/distances_parallel.pyx]: ah, so I need to wait.

File: package/MDAnalysis/lib/src/pyqcprot/pyqcprot.pyx (hunk 2 above)
Comment (hainm) [inline package/MDAnalysis/lib/src/pyqcprot/pyqcprot.pyx]: yeah, depending on taste. I prefer less typing.

**Comment by richardjgowers** [comment]
So the duplicate code in `distances.pyx` and `distances_openmp.pyx` looks like its duplicate, but it's actually different.  The header file `include/calc_distances.h` gets compiled with and without the openmp directives, so the pyx files get a different library.

**Comment by hainm** [comment]
@richardjgowers Can you point me to the line that really need to separate the pyx file? I am not really clear why?

pytraj/cpptraj can run with both serial and openmp but never bother created different pyx files.

In worse case, you should try to use IF DEF (capitalized) in Cython.

**Comment by richardjgowers** [comment]
https://github.com/MDAnalysis/mdanalysis/blob/develop/package/MDAnalysis/lib/include/calc_distances.h#L86

The point is that the ifdefs are in the header.  

You might be able to use the same .pyx for both extensions in `setup.py` actually.. I didn't think of that at the time.

**Comment by hainm** [comment]
> . I didn't think of that at the time.

:P you should update setup.py.

**Pull request closed without merging**